CLIMATE-886 Updating front-end dependencies.

Omkar20895 opened the pull request.

OCWJenkins commented:
Merged build triggered. Test Failed.

Omkar20895 commented:
I don't know how the first commit got in the way here. Anyways just try out  the package.json once and I will close this PR.

agoodm commented:
By the way in regards to the pydap tests failing: this is not on our end, but something the OPeNDAP folks need to fix. I have been in contact with their support team.

lewismc commented:
#dynamite

lewismc commented:
@Omkar20895 this is looking good. If you could just add the data to the README then I am +1 to merge into master branch.

lewismc commented:
Actually, hold off on the merge. This is not right.

lewismc commented:
Hi @Omkar20895 please merge the following into your remote branch and then test it out.
https://github.com/Omkar20895/climate/pull/1
Thanks

OCWJenkins commented:
Merged build triggered. Test Failed.

lewismc commented:
You've not addressed the README issue in the build. Also can you remove the PO.DAAC stuff?

Omkar20895 commented:
I haven't tested the branch yet, I am testing it now. I merged your PR just now into my branch and hence it appeared here.

lewismc commented:
@Omkar20895 can you delete the node dependencies folder, try again with the fresh PR applied and post your output here?

Omkar20895 commented:
@lewismc I am on a vacation so I won't be able to do it, I am gonna update this as soon as I return to my university this sunday. Thanks.

Omkar20895 commented:
@lewismc I tried this PR out in a clean virtualenv and I still got the same errors. I think we need to take others opinion if everything is working fine in your system with this PR.

@agoodm @jarifibrahim can you please try this out? Thanks :)

agoodm commented:
@Omkar20895 How are you creating the `virtualenv`? The `podaacpy` related errors that are being shown in the CI log output were fixed in 1.6.0 which is now on conda-forge. Are you certain that you are using this version in your testing?

Omkar20895 commented:
@agoodm this is not regarding travis build this is regarding the node dependencies. Please correct me if I am wrong.

lewismc commented:
@Omkar20895 have you tried deleting the npm-generated material in the ocw-ui directory and trying again?

Omkar20895 commented:
@lewismc yes I did. Just to be sure node dependencies is the only folder that npm install generates right?? If that's right then I did delete that folder and tried the patch.

jarifibrahim commented:
@Omkar20895 `npm install` would create a folder `node_modules` and install all the dependencies in it. If you do `npm install -g` then the dependencies are installed globally.

I tried your PR and got the following errors
```
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated CSSselect@0.4.1: the module is now available as 'css-select'
npm WARN deprecated CSSwhat@0.4.7: the module is now available as 'css-what'
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated tough-cookie@0.9.15: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
npm WARN deprecated minimatch@1.0.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated tough-cookie@0.12.1: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
npm WARN deprecated npmconf@2.1.2: this package has been reintegrated into npm and is now out of date with respect to npm
npm WARN prefer global coffee-script@1.10.0 should be installed with -g
npm WARN prefer global jshint@2.9.4 should be installed with -g


+-- UNMET PEER DEPENDENCY jasmine-core@*


npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.15: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN karma-jasmine@1.0.2 requires a peer of jasmine-core@* but none was installed.
```

Btw, [npm-check-update](https://www.npmjs.com/package/npm-check-updates)  package might be useful in future.

Omkar20895 commented:
@jarifibrahim yes exactly, I got the same errors and I mentioned the same earlier, according to my understanding the errors are not caused due to our part of the code, I have updated every module to the latest version.

Omkar20895 commented:
I have a theory that I have learnt from my digging up on this kind of issues, the modules that we are using in turn have unmet dependencies and their respective owners need t update them, I am not sure regarding this, please correct me if I am wrong. :)

lewismc commented:
Thats exactly right. There is nothing we can do about the deprecated transitive dependencies. Please merge in master branch.
One thing is that we need to run the tests as well
https://github.com/apache/climate/tree/master/ocw-ui/frontend/test
Can someone please open a new issue for this?

Omkar20895 commented:
Okay on it, Can you let me know what you wanted to add into the READme? I did not get that part. Thanks.

OCWJenkins commented:
Merged build triggered. Test Failed.

lewismc commented:
@Omkar20895 
http://stackoverflow.com/questions/18057466/npm-how-to-fix-no-readme-data
Please squash your commits if you could as well. Thanks, appreciated.

OCWJenkins commented:
Merged build triggered. Test Failed.

Omkar20895 commented:
@lewismc I have replicated the README in the root directory of climate and added to climate/ocw-ui/frontend. The warning was raised because npm was searching for readme data in ocw-ui. I also added a field readme in package.json, If this is alright then let me know and I will squash the commits and merge into master. Thanks :)

lewismc commented:
+1 please squash merge. Thanks

On Sun, Nov 13, 2016 at 3:57 AM, Omkar <notifications@github.com> wrote:

> @lewismc <https://github.com/lewismc> I have replicated the README in the
> root directory of climate and added to climate/ocw-ui/frontend. The warning
> was raised because npm was searching for readme data in ocw-ui. I also
> added a field readme in package.json, If this is alright then let me know
> and I will squash the commits and merge into master. Thanks :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/climate/pull/418#issuecomment-260181758>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABHJl6jhCRGWzoirUauToSFgkum_cNpwks5q9vs2gaJpZM4KiyTx>
> .
>



-- 

*Lewis*
Dr. Lewis J. McGibbney Ph.D, B.Sc
Director, MCMA Associates
Phone: +1(626)498-3090
Skype: lewis.john.mcgibbney
Email: lewis.mcgibbney@gmail.com

OCWJenkins commented:
Merged build triggered. Test Failed.
